controllers/pipelines/runconfiguration_controller_decoupled_test.go

```diff
@@ -232,9 +252,8 @@ var _ = Describe("RunConfiguration controller k8s integration", Serial, func() {
 
 			Expect(k8sClient.Status().Update(ctx, runConfiguration)).To(Succeed())
 
-			oldState := runConfiguration.Status.SynchronizationState
 			Eventually(matchRunConfiguration(runConfiguration, func(g Gomega, fetchedRc *pipelinesv1.RunConfiguration) {
-				g.Expect(fetchedRc.Status.SynchronizationState).To(Equal(oldState))
+				g.Expect(fetchedRc.Status.ObservedGeneration).To(Equal(fetchedRc.Generation))
```

Comment:
Because the state can now change to succeeded independent of dependencies, we can no longer assert on it